src/main/java/teammates/common/datatransfer/FeedbackNumericalScaleQuestionDetails.java

```diff
@@ -426,12 +425,13 @@ private boolean hasAtLeastTwoResponses(Map<String, Integer> numResponses,
         return numOfResponses != null && numOfResponses >= 2;
     }
 
-    private String getAverageExcludingSelfText(boolean showAvgExcludingSelf, DecimalFormat df, Double averageExcludingSelf) {
-        // Display a dash if the user has only self response
-        String averageExcludingSelfText = averageExcludingSelf == null ? "-" : df.format(averageExcludingSelf);
-        
-        averageExcludingSelfText = showAvgExcludingSelf ? averageExcludingSelfText : "";
-        return averageExcludingSelfText;
+    private String getAverageExcludingSelfText(boolean showAvgExcludingSelf, DecimalFormat df, Double averageExcludingSelf) {        
+        if (showAvgExcludingSelf) {
+            // Display a dash if the user has only self response
+            return averageExcludingSelf == null ? "-" : df.format(averageExcludingSelf);
+        } else {
+            return "";
+        }
```

Comment:
Surprised that PMD didn't complain here. The `else` clause might not be necessary.